src/dodal/devices/robot.py

```diff
@@ -70,8 +70,8 @@ def __init__(
         self.program_running = epics_signal_r(bool, prefix + "PROGRAM_RUNNING")
         self.program_name = epics_signal_r(str, prefix + "PROGRAM_NAME")
         self.error_str = epics_signal_r(str, prefix + "PRG_ERR_MSG")
-        # Change error_code to int type when https://github.com/bluesky/ophyd-async/issues/280 released
-        self.error_code = epics_signal_r(float, prefix + "PRG_ERR_CODE")
+        self.error_code = epics_signal_r(int, prefix + "PRG_ERR_CODE")
+        self.reset = epics_signal_x(prefix + "RESET.PROC")
```

Comment:
```suggestion
        self.reset_errors = epics_signal_x(prefix + "RESET.PROC")
```

Reply:
Thanks! From what I've gathered, `RESET.PROC` resets the state of the robot, forward linking to a lot of other resetting PVs such as `RESET_ERR_CODE` and `RESET_ALARM,` which is why I named the variable `reset` instead of `reset_errors` on the dodal level.